internal/pkg/instrumentation/bpf/net/http/client/bpf/probe.bpf.c

```diff
@@ -267,4 +174,54 @@ int uprobe_Transport_roundTrip_Returns(struct pt_regs *ctx) {
 
     bpf_map_delete_elem(&http_events, &key);
     return 0;
+}
+
+SEC("uprobe/header_writeSubset")
+int uprobe_writeSubset(struct pt_regs *ctx) {
+    u64 headers_pos = 1;
+    void *headers_ptr = get_argument(ctx, headers_pos);
+
+    u64 io_writer_pos = 3;
+    void *io_writer_ptr = get_argument(ctx, io_writer_pos);
+
+    void **key_ptr = bpf_map_lookup_elem(&http_headers, &headers_ptr);
+    if (key_ptr) {
+        void *key = *key_ptr;
+
+        struct http_request_t *http_req_span = bpf_map_lookup_elem(&http_events, &key);
+        if (http_req_span) {
+            char tp[W3C_VAL_LENGTH];
+            span_context_to_w3c_string(&http_req_span->sc, tp);
+
+            void *buf_ptr = 0;
+            bpf_probe_read(&buf_ptr, sizeof(buf_ptr), (void *)(io_writer_ptr + io_writer_buf_ptr_pos)); // grab buf ptr
+            if (!buf_ptr) {
+                goto done;
+            }
+
+            s64 size = 0;
+            bpf_probe_read(&size, sizeof(s64), (void *)(io_writer_ptr + io_writer_buf_ptr_pos + 8)); // grab size
+
+            s64 len = 0;
+            bpf_probe_read(&len, sizeof(s64), (void *)(io_writer_ptr + io_writer_n_pos)); // grab len
+
+#ifndef NO_HEADER_PROPAGATION
```

Comment:
Maybe we should put this entire function under `#ifnded`?

Reply:
Done too. I had to leave an empty stub of a probe function, so that bpf2go can compile the two versions. However I added user space code that prevents the loading of the probe.